Framework/HistogramData/src/HistogramMath.cpp

```diff
@@ -93,6 +98,55 @@ Histogram &operator-=(Histogram &histogram, const Histogram &other) {
   return histogram;
 }
 
+/// Multiplies data in other Histogram with this Histogram, propagating
+/// uncertainties.
+Histogram &operator*=(Histogram &histogram, const Histogram &other) {
+  if (histogram.yMode() == Histogram::YMode::Counts &&
+      other.yMode() == Histogram::YMode::Counts)
+    throw std::runtime_error("Invalid operation: Cannot multiply two "
+                             "histograms with YMode::Counts. Convert (one of "
+                             "them) to a distribution (YMode::Frequencies).");
+  checkSameXMode(histogram, other);
+  checkSameX(histogram, other);
+
+  auto &y1 = histogram.mutableY();
+  auto &e1 = histogram.mutableE();
+  const auto &y2 = other.y();
+  const auto &e2 = other.e();
+  for (size_t i = 0; i < y1.size(); ++i)
+    e1[i] = sqrt(pow(e1[i] * y2[i], 2) + pow(e2[i] * y1[i], 2));
+  y1 *= y2;
+  if (other.yMode() == Histogram::YMode::Counts)
+    histogram.setYMode(Histogram::YMode::Counts);
+  return histogram;
+}
+
+/// Divides data in this Histogram by other Histogram, propagating
+/// uncertainties.
+Histogram &operator/=(Histogram &histogram, const Histogram &other) {
+  if (histogram.yMode() == Histogram::YMode::Frequencies &&
+      other.yMode() == Histogram::YMode::Counts)
+    throw std::runtime_error(
+        "Invalid operation: Cannot divide histogram with YMode::Frequencies by "
+        "histogram with YMode::Counts. Convert numerator to YMode::Counts or "
+        "denominator to a distribution (YMode::Frequencies).");
+  checkSameXMode(histogram, other);
+  checkSameX(histogram, other);
+
+  auto &y1 = histogram.mutableY();
+  auto &e1 = histogram.mutableE();
+  const auto &y2 = other.y();
+  const auto &e2 = other.e();
+  for (size_t i = 0; i < y1.size(); ++i) {
+    auto inv_y2 = 1.0 / y2[i];
+    e1[i] = sqrt(pow(e1[i], 2) + pow(e2[i] * y1[i] * inv_y2, 2)) * fabs(inv_y2);
```

Comment:
Verified this matches the formula in [Divide.cpp](https://github.com/mantidproject/mantid/blob/0deb3d033a3699dcdcb9b1e548fd1f945f79356d/Framework/Algorithms/src/Divide.cpp#L47)